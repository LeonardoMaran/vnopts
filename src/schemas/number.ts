import { Schema, SchemaParameters } from '../schema';
import { Utils, ValidateResult } from '../types';

export class NumberSchema extends Schema<number, SchemaParameters<number>> {
  public expected() {
    return 'a number';
  }
  public validate(value: unknown, _utils: Utils): ValidateResult {
    return typeof value === 'number';
  }
}
