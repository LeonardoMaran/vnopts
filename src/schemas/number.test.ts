import { eachHandler } from '../../tests/__helpers__/utils';
import { NumberSchema } from './number';

const name = '<key>';
const value = 1;

describe.each`
  parameters | input                   | output               | hasWarnings
  ${{}}      | ${{ [name]: value }}    | ${{ [name]: value }} | ${false}
  ${{}}      | ${{ [name]: 'string' }} | ${Error}             | ${false}
`('', eachHandler<NumberSchema>(NumberSchema, { name }));
