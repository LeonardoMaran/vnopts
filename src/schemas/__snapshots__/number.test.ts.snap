// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema {"name":"<key>"} with input {"<key>":"string"} 1`] = `"Invalid \\"<key>\\" value. Expected a number, but received \\"string\\"."`;
